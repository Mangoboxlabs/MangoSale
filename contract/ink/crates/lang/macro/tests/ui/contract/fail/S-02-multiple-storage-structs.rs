use ink_lang as ink;

#[ink::contract]
mod multiple_storage_structs {
    #[ink(storage)]
    pub struct FirstStorageStruct {}

    // ink! currently does not allow for multiple #[ink(storage)] structs
    #[ink(storage)]
    pub struct SecondStorageStruct {}

    impl FirstStorageStruct {
        #[ink(constructor)]
        pub fn constructor1() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message1(&self) {}
    }

    impl SecondStorageStruct {
        #[ink(constructor)]
        pub fn constructor2() -> Self {
            Self {}
        }

        #[ink(message)]
        pub fn message2(&self) {}
    }
}

fn main() {}
