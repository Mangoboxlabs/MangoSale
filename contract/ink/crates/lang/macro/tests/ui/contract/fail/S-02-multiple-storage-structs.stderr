error: encountered multiple ink! storage structs, expected exactly one
  --> $DIR/S-02-multiple-storage-structs.rs:4:1
   |
4  | / mod multiple_storage_structs {
5  | |     #[ink(storage)]
6  | |     pub struct FirstStorageStruct {}
7  | |
...  |
30 | |     }
31 | | }
   | |_^

error: ink! storage struct here
 --> $DIR/S-02-multiple-storage-structs.rs:6:5
  |
6 |     pub struct FirstStorageStruct {}
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: ink! storage struct here
  --> $DIR/S-02-multiple-storage-structs.rs:10:5
   |
10 |     pub struct SecondStorageStruct {}
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
